### DISCUSSION TITLE
Should we keep the default help message?

### DISCUSSION THREAD

**Discussion by NanderTGA** [NanderTGA/msgroom-orm#151] [Ideas]
The `welcomeMessage` option has been around for a long time now and to this day has a default.
For anyone who doesn't know, this is the default: `Hi there! I'm ${client.name}. Send ${client.mainPrefix}help for a list of commands.`
This default will be used when `options.welcomeMessage` is `null` or `undefined` and the user is marked as a bot and has at least one prefix.
@nolanwhy told me utf doesn't seem to agree with this, and I think we should discuss on whether we should keep this or not.

To understand why this option has a default at all, we need to go back some time.
I needed to know the client's session ID to be able to block the current user so commands like `!repeat !repeat !repeat` wouldn't work, but wasn't passed in the `auth-complete` event yet, so I had to find a hacky workaround that involved sending a welcome message that we would then see and grab the session ID of.

**Comment by NanderTGA** [comment]
I have noticed @utf-4096 opened a PR to completely remove this option.
I don't think that would be a good change, since people can still do `client.sendMessage()` after having connected.

**Comment by utf-4096** [comment]
> I don't think that would be a good change, since people can still do client.sendMessage() after having connected.

An easily accessible option like this (which is enabled by default) incites users to keep a message.
Removing the option would make this issue be on a case-by-case basis, not the norm.

Like I said in my PR, bots should not send messages that a user didn't ask for. Bots that do this will be banned anyway, so keeping this option doesn't do much good.

**Comment by NanderTGA** [comment]
It seems to me that we should not remove the option, but mark it as deprecated and instead of sending the message, inform people about this general rule.
If you don't want welcome messages, I think the best approach is to explicitly tell people to not add them.

**Comment by utf-4096** [comment]
> I think the best approach is to explicitly tell people to not add them

I think this is common sense, regardless we already tell people not to do this whenever we notice it.

> It seems to me that we should not remove the option, but mark it as deprecated

I don't understand the point of keeping an option that will make your bot get banned, but...

**Comment by NanderTGA** [comment]
> I don't understand the point of keeping an option that will make your bot get banned, but...

I meant that I will mark the option as deprecated in the docs, and make it stop working.
When it is used, people will get a warning that they should not add welcome messages.
Like this people can be informed about this general rule, so they don't just use `client.sendMessage()` to send it anyway.

**Comment by utf-4096** [comment]
OK, sounds good.

**Comment by NanderTGA** [comment]
Changes released in `2.0.0-64`.
